### ISSUE TITLE
Generate dSYM file for flutter Plugin (iOS dynamic framework) 

### ISSUE THREAD

**Issue by zjrwdl** [flutter/flutter#94108]
Hi @mraleph @darshankawar  : 
There is a similar case with https://github.com/flutter/flutter/issues/87320
How to generate dSYM file for flutter plugin dynamic framework?**_(config use_framework in podfile to make plugin dynamic framework)_**
" --extra-gen-snapshot-options=--no-strip " only works for App.framework.
As i build ios framework with command "flutter build ios-framework --no-tree-shake-icons --no-debug --no-profile" plugin dynamic framework and dSYM file were both generated , but dSYM file soon disappeard.Without dSYM file , crash stack may not be translated.
I guess it may be deleted by the system config. Do you know where is it?
Looking forward to your reply！Thanks !

**Comment by maheshmnj** [comment]
Keeping this issue open for further insights from the team.

**Comment by jmagman** [comment]
> plugin dynamic framework and dSYM file were both generated , but dSYM file soon disappeard

What does it mean it disappeared?  You ran the command and it was generated, so where did it go?

**Comment by zjrwdl** [comment]
> > plugin dynamic framework and dSYM file were both generated , but dSYM file soon disappeard
> 
> What does it mean it disappeared? You ran the command and it was generated, so where did it go?

I guess dSYM file was deleted after it was generated. 
For users, what they see is that there is no generation.
Do you know , how to get dSYM?

**Comment by jmagman** [comment]
> I guess dSYM file was deleted as default config after it was generated.

Can you give us reproduction steps?  If `flutter` is generating the file I don't understand what is deleting it, and when.

**Comment by zjrwdl** [comment]
> > I guess dSYM file was deleted as default config after it was generated.
> 
> Can you give us reproduction steps? If `flutter` is generating the file I don't understand what is deleting it, and when.

1. write a simple flutter plugin project 
2. write use_framework in podfile to make plugin ios dynamic framework(static framework as default)。
3. run command "flutter build ios-framework --no-tree-shake-icons --no-debug --no-profile"

**Comment by jmagman** [comment]
I see the dSYMs in the XCFramework, they don't 
```
example/build/ios/framework/Release/test_plugin.xcframework/ios-arm64_armv7/dSYMs/test_plugin.framework.dSYM
```
Do you not see them there?

What's the output of `flutter doctor -v`?

**Comment by zjrwdl** [comment]
> I see the dSYMs in the XCFramework, they didn't disappear:
> 
> ```
> example/build/ios/framework/Release/test_plugin.xcframework/ios-arm64_armv7/dSYMs/test_plugin.framework.dSYM
> ```
> 
> Do you not see them there?
> 
> What's the output of `flutter doctor -v`?

Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.2.0, on macOS 11.2.3 20D91 darwin-x64, locale zh-Hans-CN)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.2)
[✓] IntelliJ IDEA Ultimate Edition (version 2020.2.4)
[✓] Connected device (2 available)

• No issues found!

**Comment by jmagman** [comment]
> May you make a try by using flutter 2.2.0 ?

Duplicate of https://github.com/flutter/flutter/issues/80063.  That fix is available in 2.3.0-0.0.pre.

**Comment by zjrwdl** [comment]
> available

thanks very much
